src/Language/PureScript/Errors.hs

```diff
@@ -250,12 +252,55 @@ errorMessage err = MultipleErrors [err]
 onErrorMessages :: (ErrorMessage -> ErrorMessage) -> MultipleErrors -> MultipleErrors
 onErrorMessages f = MultipleErrors . map f . runMultipleErrors
 
+replaceUnknowns :: Type -> State [Unknown] Type
+replaceUnknowns = everywhereOnTypesTopDownM replaceUnknowns'
+  where
+    lookupTable :: Unknown -> [Unknown] -> (Unknown, [Unknown])
+    lookupTable x table = case (findIndex (== x) table) of
```

Comment:
Actually `elemIndex` is the one you want, sorry.

Also, redundant parens here.